068-performance-testing.md

```diff
@@ -0,0 +1,451 @@
+It allows developers and testers to simulate traffic and user behavior towards applications, APIs,
+and services to identify potential bottlenecks and performance limits under controlled test environments.
+k6 integrates seamlessly with [Grafana](https://grafana.com/), a popular open-source analytics and monitoring solution,
+enabling users to visualize test results and performance metrics in real-time through comprehensive dashboards.
+This integration facilitates easy analysis and interpretation of data, making it simpler for teams to make informed
+decisions based on the performance characteristics of their applications.
+
+One of the key features of k6 is its extensibility through extensions, which allows users to tailor the tool to their specific testing needs.
+For instance, the ability to extend k6's functionality to work with specific technologies like Kubernetes through
+the xk6-kubernetes extension showcases its flexibility.
+Similarly, this extensibility opens up the possibility of developing a dedicated extension for Strimzi, enabling specialized
+performance testing for Strimzi-managed Apache Kafka clusters.
+This approach would allow for targeted load testing of Kafka operators, providing insights into how Strimzi components
+behave under various load conditions, thereby ensuring that Strimzi deployments are optimized for performance and reliability.
+
+The primary challenge is that developing such an extension requires proficiency in Go/Javascript,
+which might be less familiar to our community accustomed to Java.
+
+####  Notes on Release Cycle for Using k6:
+
+The release cycle for incorporating k6 into Strimzi testing would bear similarities to the dedicated repository approach,
+with both pros and cons.
+One advantage is the potential for rapid adoption and integration of new performance testing methodologies,
+given k6's established user base and continuous updates.
+However, a possible downside is the need to maintain compatibility with Strimzi's evolving infrastructure,
+possibly requiring frequent updates to any Strimzi-specific extensions or configurations within k6.
+
+## Selected Approach/Proposal: Approach 1 - Integration within the strimzi-kafka-operator
+
+We propose integrating performance testing directly alongside existing system tests, leveraging the established infrastructure and testing utilities. 
+This method ensures that performance and system tests can share resources efficiently, streamlines the development process, facilitates immediate feedback on performance impacts, and reduces overhead.
+
+- This method integrates performance testing directly alongside existing system tests. 
+- It leverages the already established infrastructure and testing utilities, ensuring that performance and system tests can share resources such as the ResourceManager, LogCollector, and MetricsCollector.
```

Comment:
These two things are already mentioned on the previous two lines or?

Reply:
Yeah, nice catch :)